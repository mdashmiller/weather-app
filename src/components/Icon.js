import styled from "styled-components"

const Icon = styled.i`
    display: inline-block;
    width: 40px;
    margin: 10px;
    color: #c3c3c3;
`

export default Icon
