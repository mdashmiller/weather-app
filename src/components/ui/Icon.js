import styled, { css } from "styled-components"

const Icon = styled.i`
    display: inline-block;
    width: 40px;
    margin: 20px;
    color: #c3c3c3;

    ${props => props.dark && css`
        color: #818181;
      `}
`

export default Icon
