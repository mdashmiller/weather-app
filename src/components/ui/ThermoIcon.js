import React from 'react'
import Icon from './Icon'

// weather-dependent icon details
const thermoHot = "fas fa-thermometer-full fa-3x"
const thermoWarm = "fas fa-thermometer-three-quarters fa-3x"
const thermoCool = "fas fa-thermometer-half fa-3x"
const thermoCold = "fas fa-thermometer-empty fa-3x"

const ThermoIcon = ({ temp }) => {
	// determines which thermometer icon to
	// display dependent upon temperature
	if (temp > 32 && temp < 69) {
		return <Icon className={thermoCool}></Icon>
	} else if (temp >= 69 && temp < 86) {
		return <Icon className={thermoWarm}></Icon>
	} else if (temp >= 86) {
		return <Icon className={thermoHot}></Icon>
	} else {
		return <Icon className={thermoCold}></Icon>
	}
}

export default ThermoIcon
