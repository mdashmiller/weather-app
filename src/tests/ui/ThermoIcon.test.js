import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ThermoIcon from '../../components/ui/ThermoIcon'
import Icon from '../../components/ui/Icon'

Enzyme.configure({ adapter: new Adapter })

describe('<ThermoIcon /> rendering', () => {

	it('renders without crashing', () => {
	  	const div = document.createElement('div')
	  	ReactDOM.render(<ThermoIcon />, div)
	  	ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(<ThermoIcon />)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

})

describe('<ThermoIcon /> displays proper className for each temp category', () => {

	// icon varieties
	const thermoHot = "fas fa-thermometer-full fa-3x"
	const thermoWarm = "fas fa-thermometer-three-quarters fa-3x"
	const thermoCool = "fas fa-thermometer-half fa-3x"
	const thermoCold = "fas fa-thermometer-empty fa-3x"

	it('has className {thermoCool} when temp > 32 && temp < 69', () => {
		const wrapper = shallow(<ThermoIcon temp={50} />)

		expect(wrapper.find(Icon).prop('className')).toBe(thermoCool)
	})

	it('has className {thermoWarm} when temp >= 69 && temp < 86', () => {
		const wrapper = shallow(<ThermoIcon temp={75} />)

		expect(wrapper.find(Icon).prop('className')).toBe(thermoWarm)
	})

	it('has className {thermoHot} when temp >= 86', () => {
		const wrapper = shallow(<ThermoIcon temp={90} />)

		expect(wrapper.find(Icon).prop('className')).toBe(thermoHot)
	})

	it('has className {thermoCold} when there is no temp prop', () => {
		const wrapper = shallow(<ThermoIcon />)
		
		expect(wrapper.find(Icon).prop('className')).toBe(thermoCold)
	})

})
