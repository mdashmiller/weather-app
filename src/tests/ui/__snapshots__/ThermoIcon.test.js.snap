// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThermoIcon /> rendering has a valid snapshot 1`] = `
<i
  className="fas fa-thermometer-empty fa-3x sc-bdVaJa bSIgLK"
/>
`;
